### ISSUE TITLE
Certificate handling for Linux WebKit ports

### ISSUE THREAD

**Issue by clopez** [web-platform-tests/wpt#19233]
The current driver for webkitgtk/epiphany/wpewebkit passes the wpt certificate path to the WebKitWebDriver in a message with the format

`{"host": "domain.test", "certificateFile": "/path/to/cert"}`

Then WebKit adds this certificate but only for that domain. It doesn't add the certificate to the trusted DB of certificates. It just adds an exception for that host to validate that cert. Check https://trac.webkit.org/changeset/233035

So the list of hosts for which the certificate is valid has to be generated manually and passed on the previous webdriver message.

The issue is that the current browser driver define that domain to be just "web-platform.test", which is not enough as this causes failures on tests that try to load resources via https from other subdomains like "www1.web-platform.test". Example of such test: /cookies/samesite/fetch.https.html

/cc @psaavedra @zdobersek

**Comment by clopez** [comment]
Another option may be to set `"acceptInsecureCerts": True` in the browser capabilities, but maybe that is too excessive and there are tests that check that a cert is invalid for a given site in order to pass?

**Comment by psaavedra** [comment]
@clopez, take a look on this:
https://github.com/web-platform-tests/wpt/pull/19185. This PR install the
WPT CA certificates in the system fixing the insecure certificate issues.

**Comment by clopez** [comment]
> @clopez, take a look on this: #19185. This PR install the WPT CA certificates in the system fixing the insecure certificate issues.

Right, that fixes the issue by installing the certificate in the container running the tests on the CI  (taskcluster).

However the method discussed here is different, because it allows to run this tests without installing any certificate, so its ideal for developers or for anyone testing with wpt locally, because it doesn't require them to install any insecure certificate. So its complementary to that.

**Comment by foolip** [comment]
Does it work if all of the host names listed by `./wpt make-hosts-file` are given?

**Comment by clopez** [comment]
> Does it work if all of the host names listed by `./wpt make-hosts-file` are given?

I guess it would work... but is the certificate really needed for all those hosts?

The shipped certificate seems to be valid only for this 12 hosts:
```
$ cat ./tools/certs/cacert.pem | openssl x509 -text -noout|grep -Po DNS:[^,]+|cut -d: -f2-|sort|uniq
not-web-platform.test
web-platform.test
www1.not-web-platform.test
www1.web-platform.test
www2.not-web-platform.test
www2.web-platform.test
www.not-web-platform.test
www.web-platform.test
xn--lve-6lad.not-web-platform.test
xn--lve-6lad.web-platform.test
xn--n8j6ds53lwwkrqhv28a.not-web-platform.test
xn--n8j6ds53lwwkrqhv28a.web-platform.test
```

Not sure if there is an easy way of getting this list that doesn't involve parsing the certificate itself

**Comment by foolip** [comment]
@gsnedders what's the internal way to get a list of hosts that the cert should cover?

**Comment by gsnedders** [comment]
Pretty sure in theory the certificate should cover everything. Given regenerating it has historically been painful I'm not surprised if it doesn't match up!

In principle, running with `acceptInsecureCerts` should be fine (Chrome always does, Firefox does if there's no `certutil` available). Otherwise, `server_config.domains_set` should be a set of all the domains it should cover.

**Comment by gsnedders** [comment]
>Pretty sure in theory the certificate should cover everything. Given regenerating it has historically been painful I'm not surprised if it doesn't match up!

 #19626 to update them

**Comment by clopez** [comment]
> Otherwise, `server_config.domains_set` should be a set of all the domains it should cover.

That works as expected. I have opened PR #20213 with a patch that implements passing that list.

**Comment by clopez** [comment]
Fixed in 513f82848f14ce777702a819f91c9f41c132af91
